applications/FluidDynamicsApplication/python_scripts/navier_stokes_two_fluids_solver.py

```diff
@@ -0,0 +1,269 @@
+                                                                            self.settings["reform_dofs_at_each_step"].GetBool(),
+                                                                            self.settings["move_mesh_flag"].GetBool())
+
+        (self.solver).SetEchoLevel(self.settings["echo_level"].GetInt())
+
+        (self.solver).Initialize() # Initialize the solver. Otherwise the constitutive law is not initializated.
+        (self.solver).Check()
+
+        self.main_model_part.ProcessInfo.SetValue(KratosMultiphysics.DYNAMIC_TAU, self.settings["dynamic_tau"].GetDouble())
+
+        KratosMultiphysics.Logger.PrintInfo("NavierStokesTwoFluidsSolver", "Solver initialization finished.")
+
+    def InitializeSolutionStep(self):
+        if self._TimeBufferIsInitialized():
+            (self.bdf_process).Execute()                    # Recompute the BDF2 coefficients
+            (self.level_set_convection_process).Execute()   # Convect the level-set according to the previous step velocity
+            (self.variational_distance_process).Execute()   # Recompute the distance field according to the new level set position
+            self._set_physical_properties()                 # Update the DENSITY and DYNAMIC_VISCOSITY values according to the new level set
+            (self.solver).InitializeSolutionStep()          # Initialize the solver current step
+
+    def FinalizeSolutionStep(self):
+        if self._TimeBufferIsInitialized():
+            (self.solver).FinalizeSolutionStep()
+            (self.accelerationLimitationUtility).Execute()
+
+
+    def _set_physical_properties(self):
+        # Get fluid 1 and 2 properties
+        properties_1 = self.main_model_part.Properties[1]
+        properties_2 = self.main_model_part.Properties[2]
+
+        rho_1 = properties_1.GetValue(KratosMultiphysics.DENSITY)
+        rho_2 = properties_2.GetValue(KratosMultiphysics.DENSITY)
+        mu_1 = properties_1.GetValue(KratosMultiphysics.DYNAMIC_VISCOSITY)
+        mu_2 = properties_2.GetValue(KratosMultiphysics.DYNAMIC_VISCOSITY)
+
+        if rho_1 <= 0.0:
+            raise Exception("DENSITY set to {0} in Properties {1}, positive number expected.".format(rho_1, properties_1.Id))
+        if rho_2 <= 0.0:
+            raise Exception("DENSITY set to {0} in Properties {1}, positive number expected.".format(rho_2, properties_2.Id))
+        if mu_1 <= 0.0:
+            raise Exception("DYNAMIC_VISCOSITY set to {0} in Properties {1}, positive number expected.".format(mu_1, properties_1.Id))
+        if mu_2 <= 0.0:
+            raise Exception("DYNAMIC_VISCOSITY set to {0} in Properties {1}, positive number expected.".format(mu_2, properties_2.Id))
+
+        # Transfer density and (dynamic) viscostity to the nodes
+        for node in self.main_model_part.Nodes:
+            if node.GetSolutionStepValue(KratosMultiphysics.DISTANCE) < 0.0:
+                node.SetSolutionStepValue(KratosMultiphysics.DENSITY, rho_1)
```

Comment:
This should be distance <= 0.0 to be consistent with the rest of the code,

Reply:
Ok. Done.